src/main/scala/org/broadinstitute/dsde/agora/server/webservice/routes/RouteHelpers.scala

```diff
@@ -0,0 +1,135 @@
+package org.broadinstitute.dsde.agora.server.webservice.routes
+
+import akka.actor.Props
+import org.broadinstitute.dsde.agora.server.model.{AgoraEntityType, AgoraEntityProjection, AgoraEntity}
+import org.broadinstitute.dsde.agora.server.webservice.util.ServiceMessages.{Add, Query, QuerySingle}
+import org.broadinstitute.dsde.agora.server.webservice.PerRequestCreator
+import spray.routing.{Directive0, Directives, RequestContext}
+
+import scala.util.Try
+
+
+trait RouteHelpers extends AgoraDirectives
+  with QuerySingleRoute
+  with QueryRoute
+  with AddRoute
+
+trait BaseRoute extends PerRequestCreator with RouteUtil
+
+trait QuerySingleRoute extends BaseRoute {
```

Comment:
Hmm, personally, I think there is enough context: the file and parent traits are both called RouteHelpers. And the trait keyword suggests that you can find the concrete implementation elsewhere. What do other people think? @dshiga @gbggrant @jacarey